### PR TITLE
Add Hexagon arch and mach

### PR DESCRIPTION
**Pull request by Rot127**
Adds the Hexagon architecture enum entry and defines for machine types.

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
@Rot127 you need to update also the `lib/frame_arch.ml`

**Comment by Rot127** [comment]
Ah right, will take care of it in August.

**Review by XVilka — APPROVED** [review]

File: libtrace/src/frame_arch.h [hunk 1]
```diff
@@ -385,6 +385,42 @@ enum frame_architecture
   frame_arch_aarch64,   /* AArch64.  */
 #define frame_mach_aarch64 0
   frame_arch_8051,/* Intel 8051. */
+  frame_arch_hexagon, /* Qualcomm DSP architecture Hexagon */
```
Comment (thestr4ng3r) [inline libtrace/src/frame_arch.h:388]: Please put this after the `frame_arch_sm83` (as well as in `frame_arch.ml`) so compatibility with previous commits on master is preserved.

**Review by thestr4ng3r — CHANGES_REQUESTED** [review]

**Review by thestr4ng3r — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Review by ivg — APPROVED** [review]
👍🏻

**Pull request merged**